hibernate-core/src/main/java/org/hibernate/loader/Loader.java

```diff
@@ -1896,6 +1897,25 @@ private void loadFromResultSet(
 				session.getPersistenceContextInternal().addEntity( euk, object );
 			}
 		}
+		boolean[] propertyUniqueness = ( (AbstractEntityPersister) persister ).getPropertyUniqueness();
+		for ( int index = 0; index < propertyUniqueness.length; index++ ) {
+			if ( propertyUniqueness[index] ) {
+				final Type type = persister.getPropertyTypes()[index];
+				final Object uniqueKey = type.semiResolve( values[index], session, object );
```

Comment:
One thing to consider is that in the case of a property-ref that contains a many-to-one, that many-to-one may get initialized, because [`ComponentType#semiResolve`](https://github.com/hibernate/hibernate-orm/blob/main/hibernate-core/src/main/java/org/hibernate/type/ComponentType.java#L709-L714) will delegate to `#resolve`.

I'm not sure if this is a real issue. Please add test case that has a property-ref with a lazy many-to-one association, so we can see what happens.